astarte/device/device.py

```diff
@@ -156,6 +157,59 @@ def add_interface(self, interface_definition: dict) -> None:
         """
         self.__introspection.add_interface(interface_definition)
 
+    def add_interface_from_file(self, interface_file: Path):
+        """
+        Adds an interface to the device
+
+        This will add an interface definition to the device. It has to be called before
+        :py:func:`connect`, as it will be used for building the device introspection.
+
+        Parameters
+        ----------
+        interface_file : Path
+            An absolute path to an Astarte interface json file.
+        Raises
+        ------
+        FileNotFoundError
+            If specified file does not exists.
+        TypeError
+            If speficied file is not a .json file.
+        """
+        if not interface_file.is_file():
+            raise FileNotFoundError(f'"{interface_file}" does not exist or is not a file')
+        if interface_file.suffix != ".json":
+            raise TypeError(f'"{interface_file}" is not a json file')
+        with open(interface_file, "r", encoding="utf-8") as interface_fp:
+            self.__introspection.add_interface(json.load(interface_fp))
+
+    def add_interfaces_from_dir(self, interfaces_dir: Path):
+        """
+        Adds a series of interfaces to the device
+
+        This will add all the interfaces contained in the provided folder to the device.
+        It has to be called before :py:func:`connect`, as it will be used for building the device
+        introspection.
+
+        Parameters
+        ----------
+        interfaces_dir : Path
+            An absolute path to an a folder containing some Astarte interface .json files.
+        Raises
+        ------
+        FileNotFoundError
+            If specified directory does not exists.
+        NotADirectoryError
+            If speficied directory exists but it's not a directory.
+        """
+        if not interfaces_dir.exists():
+            raise FileNotFoundError(f'"{interfaces_dir}" does not exist')
+        if not interfaces_dir.is_dir():
+            raise NotADirectoryError(f'"{interfaces_dir}" is not a directory')
+        for interface_file in interfaces_dir.iterdir():
+            if interface_file.is_file() and (interface_file.suffix == ".json"):
+                with open(interface_file, "r", encoding="utf-8") as interface_fp:
+                    self.__introspection.add_interface(json.load(interface_fp))
```

Comment:
```suggestion
                self.add_interface_from_file(interface_file)
```

Reply:
Arguably I'd add this to the "Raises" of the docstring if pylint is ok with it.
``` 
TypeError
            If the specified directory contains a file with the .json suffix but is not a JSON.
```

Reply:
I agree with calling `add_interface_from_file` instead of doing the operation directly.
Regarding the second comment. Starting to list also exceptions raised by subfunctions might lead to a couple of problems:
- The reader to think that all sub-functions "Raises" are documented in the docstring. This is difficult to maintain. When adding a new `raise` clause to a function we would have to search for all callers to such function and change the docstrings of each one. 
- Built-in functions and functions from external modules often raise exceptions, propagating this would be hard to maintain.